zio-http-datastar-sdk/src/main/scala/zio/http/datastar/ServerSentEventGenerator.scala

```diff
@@ -0,0 +1,195 @@
+package zio.http.datastar
+
+import zio._
+
+import zio.http.ServerSentEvent
+import zio.http.template2._
+
+sealed trait EventType extends Product with Serializable
+
+object EventType {
+  case object PatchElements extends EventType {
+    override def toString: String = "datastar-patch-elements"
+  }
+  case object PatchSignals  extends EventType {
+    override def toString: String = "datastar-patch-signals"
+  }
+}
+
+sealed trait ElementPatchMode extends Product with Serializable {
+  override def toString: String = this.productPrefix.toLowerCase
+}
```

Comment:
Same suggestion as https://github.com/zio/zio-http/pull/3649/files#r2297080439

```suggestion
sealed trait ElementPatchMode extends Product with Serializable {
  final def render: String =
    this match {
      case ElementPatchMode.Outer   => "outer"
      case ElementPatchMode.Inner   => "inner"
      case ElementPatchMode.Replace => "replace"
      case ElementPatchMode.Prepend => "prepend"
      case ElementPatchMode.Append  => "append"
      case ElementPatchMode.Before  => "before"
      case ElementPatchMode.After   => "after"
      case ElementPatchMode.Remove  => "remove"
    }
}
```

later:
```scala
    // instead of `options.mode.toString` which is fragile in terms of refactoring
    if (options.mode != ElementPatchMode.Outer) {
      sb.append("mode ").append(options.mode.render).append("\n")
    }
```